server/src/test/java/org/elasticsearch/cluster/routing/allocation/shards/ShardsAvailabilityHealthIndicatorServiceTests.java

```diff
@@ -1913,6 +2003,88 @@ public void testMappedFieldsForTelemetry() {
+        ShardId id = new ShardId("index", "uuid", 0);
+        IndexMetadata idxMeta = IndexMetadata.builder("index")
+            .numberOfShards(1)
+            .numberOfReplicas(1)
+            .settings(
+                Settings.builder()
+                    .put("index.number_of_shards", 1)
+                    .put("index.number_of_replicas", 1)
+                    .put("index.version.created", IndexVersion.current())
+                    .put("index.uuid", "uuid")
+                    .build()
+            )
+            .build();
+        ShardRouting primary = createShardRouting(id, true, new ShardAllocation("node", AVAILABLE));
+        var state = createClusterStateWith(List.of(index("index", new ShardAllocation("node", AVAILABLE))), List.of());
+        assertFalse(ShardsAvailabilityHealthIndicatorService.isNewlyCreatedAndInitializingReplica(primary, state));
+
+        ShardRouting replica = createShardRouting(id, false, new ShardAllocation("node", AVAILABLE));
+        state = createClusterStateWith(List.of(index("index", new ShardAllocation("node", AVAILABLE))), List.of());
+        assertFalse(ShardsAvailabilityHealthIndicatorService.isNewlyCreatedAndInitializingReplica(replica, state));
+
+        ShardRouting unassignedReplica = createShardRouting(id, false, new ShardAllocation("node", UNAVAILABLE));
+        state = createClusterStateWith(
+            List.of(idxMeta),
+            List.of(index("index", "uuid", new ShardAllocation("node", UNAVAILABLE))),
+            List.of(),
+            List.of()
+        );
+        assertFalse(ShardsAvailabilityHealthIndicatorService.isNewlyCreatedAndInitializingReplica(unassignedReplica, state));
+
+        ShardAllocation allocation = new ShardAllocation(
+            "node",
+            UNAVAILABLE,
+            new UnassignedInfo(
+                UnassignedInfo.Reason.NODE_LEFT,
+                "message",
+                null,
+                0,
+                0,
+                0,
+                randomBoolean(),
+                UnassignedInfo.AllocationStatus.DECIDERS_NO,
+                Set.of(),
+                null
+            )
+        );
+        ShardRouting decidersNoReplica = createShardRouting(id, false, allocation);
+        state = createClusterStateWith(
+            List.of(idxMeta),
+            List.of(index(idxMeta, new ShardAllocation("node", UNAVAILABLE), allocation)),
+            List.of(),
+            List.of()
+        );
+        assertFalse(ShardsAvailabilityHealthIndicatorService.isNewlyCreatedAndInitializingReplica(decidersNoReplica, state));
+
+        allocation = new ShardAllocation(
+            "node",
+            UNAVAILABLE,
+            new UnassignedInfo(
+                UnassignedInfo.Reason.NODE_LEFT,
+                "message",
+                null,
+                0,
+                0,
+                0,
+                randomBoolean(),
+                UnassignedInfo.AllocationStatus.NO_ATTEMPT,
```

Comment:
I was curious why the value of this here and above were the only differences. I replaced them locally with a random value -- it doesn't seem to make any difference to the test. Is that expected? Is it just accidental that they're different?

Reply:
Just accidental, these tests went through a lot of iterations so it's probably a holdover from that. The real difference is the `CREATING` state below

Reply:
OK, so the value of the AllocationStatus does not matter? It might make the test cleaner to either make it random or have a comment that it doesn't matter (I got hung up on trying to understand the difference in the two `allocation`s for a while).